src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -100,6 +102,74 @@ void SHAMap::visitNodes(std::function<bool (SHAMapAbstractNode&)> const& functio
     }
 }
 
+void
+SHAMap::visitDifferences(SHAMap const* have,
+    std::function<bool (SHAMapAbstractNode&)> function) const
+{
+    // Visit every node in this SHAMap that is not present
+    // in the specified SHAMap
+    assert (root_->isValid ());
+
+    if (! root_)
+        return;
```

Comment:
Hmm... I see we that this issue already existing here:  https://github.com/ripple/rippled/blob/develop/src/ripple/shamap/impl/SHAMapSync.cpp#L43-L46
Perhaps they should be reversed in both places.